modules/internal/ChapelBase.chpl

```diff
@@ -1286,23 +1306,29 @@ module ChapelBase {
     __primitive("/=", lhs, rhs);
   }
   inline proc /=(ref lhs:real(?w), rhs:real(w)) {
+    if (chpl_checkDivByZero) then
+      if rhs == 0.0 then
+        halt("Attempt to divide by zero");
```

Comment:
Unfortunately, it looks like the ``/(param a: uint(64), b: uint(64))`` overload of division is currently required for expressions like "10/myUint" due to a longstanding issue with param coercion for uints that's been in my court for a long time.  In the interest of keeping your PR simple, I wouldn't worry about adding the param/non-param overload for % since we don't have it today...  though you also could do so in order to improve symmetry between the two.